analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/IOperationExtensionsTest.cs

```diff
@@ -152,32 +151,58 @@ public void TrackedSymbol_DeclarationExpression(string assignment)
     {
         var code = $"public class C {{ void Method() {{ {assignment}; }} }}";
         var graph = TestHelper.CompileCfgCS(code);
-        var allDeclarations = graph.Blocks[1].Operations.SelectMany(x => x.DescendantsAndSelf()).Where(x => x.Kind == OperationKindEx.DeclarationExpression).Select(IDeclarationExpressionOperationWrapper.FromOperation).ToArray();
+        var allDeclarations = graph.Blocks[1].Operations
+            .SelectMany(x => x.DescendantsAndSelf()).Where(x => x.Kind == OperationKindEx.DeclarationExpression).Select(IDeclarationExpressionOperationWrapper.FromOperation).ToArray();
         allDeclarations.Should().NotBeEmpty();
         allDeclarations.Should().AllSatisfy(x =>
-            x.WrappedOperation.TrackedSymbol().Should().NotBeNull().And.BeAssignableTo<ISymbol>()
+            x.WrappedOperation.TrackedSymbol(ProgramState.Empty).Should().NotBeNull().And.BeAssignableTo<ISymbol>()
             .Which.GetSymbolType().Should().NotBeNull().And.BeAssignableTo<ITypeSymbol>()
             .Which.SpecialType.Should().Be(SpecialType.System_Int32));
     }
 
+    [TestMethod]
+    public void TrackedSymbol_CaptureReference()
+    {
+        var cfg = TestHelper.CompileCfgBodyCS("a ??= b;", "object a, object b");
+        var capture = IFlowCaptureOperationWrapper.FromOperation(cfg.Blocks[1].Operations[0]);
+        var captureReference = cfg.Blocks[3].Operations[0].ChildOperations.First();
+        var state = ProgramState.Empty.SetCapture(capture.Id, capture.Value);
+        captureReference.TrackedSymbol(state).Name.Should().Be("a");
+    }
+
+    [TestMethod]
+    public void TrackedSymbol_CaptureReferenceInConversion()
+    {
+        var cfg = TestHelper.CompileCfgBodyCS("_ = (string)(condition ? a : null);", "object a, bool condition");
+        var capture = IFlowCaptureOperationWrapper.FromOperation(cfg.Blocks[2].Operations[0]);
+        var conversion = cfg.Blocks[4].Operations[0].ChildOperations.First().ChildOperations.Skip(1).First();
+        var state = ProgramState.Empty.SetCapture(capture.Id, capture.Value);
+        conversion.TrackedSymbol(state).Name.Should().Be("a");
+    }
+
+    [TestMethod]
+    public void TrackedSymbol_NullSafe() =>
+        (null as IOperation).TrackedSymbol(ProgramState.Empty).Should().BeNull();
```

Comment:
I'd do it like this, the `as` has some additional logic, because it should have been `((IOperation)null)` instead (stupid C# casts)
```suggestion
        IOperationExtensions.TrackedSymbol(null, ProgramState.Empty).Should().BeNull();
```